packages/web/src/layouts/earn/containers/pool-list-container/PoolListContainer.tsx

```diff
@@ -34,143 +34,172 @@ const PoolListContainer: React.FC = () => {
 
   const themeKey = useAtomValue(ThemeState.themeKey);
 
+  /**
+   * Checks if either token has missing price information
+   */
   const anyEmptyPrice = useCallback(
     (tokenA: TokenModel, tokenB: TokenModel) =>
       !tokenPrices?.[checkGnotPath(tokenA.priceID)]?.usd || !tokenPrices?.[checkGnotPath(tokenB.priceID)]?.usd,
     [tokenPrices],
   );
 
+  /**
+   * Fetch pool data on component mount
+   */
   useEffect(() => {
     updatePools();
   }, []);
 
+  /**
+   * Hide search icon when clicking outside and no keyword is entered
+   */
   useEffect(() => {
-    if (!keyword) {
-      if (isClickOutside) {
-        setSearchIcon(false);
-      }
+    if (!keyword && isClickOutside) {
+      setSearchIcon(false);
     }
   }, [isClickOutside, keyword]);
 
+  /**
+   * Reset page to first page when filter criteria change
+   */
+  useEffect(() => {
+    setPage(0);
+  }, [keyword, poolType]);
+
+  /**
+   * Format pool value with proper formatting or fallback to "-" when price data is missing
+   */
+  const formatPoolValue = useCallback(
+    (value: string | number | undefined, tokenA: TokenModel, tokenB: TokenModel) => {
+      if (anyEmptyPrice(tokenA, tokenB)) return "-";
+
+      return formatOtherPrice(value || 0, {
+        isKMB: false,
+        decimals: 0,
+      });
+    },
+    [anyEmptyPrice],
+  );
+
+  /**
+   * Check if pool info matches the search keyword
+   */
+  const matchesKeyword = useCallback((info: PoolListInfo, keyword: string) => {
+    if (!keyword) return true;
+
+    const searchTerm = keyword.toLowerCase();
+    return (
+      info.tokenA.name.toLowerCase().includes(searchTerm) ||
+      info.tokenB.name.toLowerCase().includes(searchTerm) ||
+      info.tokenA.symbol.toLowerCase().includes(searchTerm) ||
+      info.tokenB.symbol.toLowerCase().includes(searchTerm)
+    );
+  }, []);
+
+  /**
+   * Transform string values to numbers for sorting
+   */
   const sortValueTransform = (value: string) => {
```

Comment:
consider specifying the return type.
```suggestion
  const sortValueTransform = (value: string): number => {
```

Reply:
Reflected